core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/ApplyUserAgentStage.java

```diff
@@ -94,6 +96,15 @@ private StringBuilder getUserAgent(SdkClientConfiguration config, List<ApiName>
                  .append("/")
                  .append(SdkHttpUtils.urlEncode(clientName));
 
+        String retryMode = config.option(SdkClientOption.RETRY_POLICY).retryMode().toString();
```

Comment:
Can retry mode be null? E.g. if the customer does `RetryPolicy.builder((RetryMode) null).build()`? Arguably that should probably not be allowed, but I feel like it can happen.

Reply:
It will trigger another exception while the retry mode is being used, but to make it more clear that the retry mode should not be set as null, a new exception is added to the `RetryPolicy` class.